static/js/syntaxModesRules.ts

```diff
@@ -8,13 +8,16 @@ import {LANG_tr} from './syntaxLang-tr';
 import {LANG_id} from './syntaxLang-id';
 import {LANG_nb_NO} from './syntaxLang-nb_NO';
 
+// #/#/@/#/# We don't want to keep this block of code for automatic test execution
```

Comment:
How about, instead of post-processing this file at a textual level to remove the parts you don't want to test, you factor out the parts you DO want to test?

So, instead of doing this:

```
┌───────────────────────┐                        ┌───────────────────────┐
│                       │                        │                       │
│  syntaxModesRules.ts  │───────────────────────▶│      testable.ts      │
│                       │      ┌─────────┐       │                       │
└───────────────────────┘      │script.py│       └───────────────────────┘
                               └─────────┘                                
```

Do this instead:

```
            ┌─────────────────────────────────┐            
            │                                 │            
            │        essenceOfRules.ts        │            
            │                                 │            
            └─────────────────────────────────┘            
            ▲                                 ▲            
            │                                 │            
     import │                                 │  import    
            │                                 │            
            │                                 │            
┌───────────────────────┐         ┌───────────────────────┐
│                       │         │                       │
│  syntaxModesRules.ts  │         │      testable.ts      │
│                       │         │                       │
└───────────────────────┘         └───────────────────────┘
```

Reply:
Perhaps an even better way, instead of generating `data.json` solely for testing, is to make it the official intermediate format:

- It gets generated at dev time from whatever convenient (Python?) script you all care to write. That way, the syntax highlighting rules don't have to live in TypeScript.
- The tests can load it and simulate the syntax highlighting according to the rules.
- It gets bundled into the `.js` bundle by `esbuild`-- in TypeScript you'll be able to do `import rules_en from './en.json'` and have the JSON object available, ready for passing into Ace.

Like this:

```
┌──────────────────────────┐                                                                                    
│                          │                                                                                    
│ genenerateHighlights.py  │                                                                                    
│                          │                                                                                    
└──────────────────────────┘           ┌─────────────────────────────────┐                                      
              │                        │             en.json             │      import as data                  
              └───────────────────────▶│             nl.json             │◀────────────────────────┐            
                      generate at      │             de.json             │                         │            
                       dev time        └─────────────────────────────────┘             ┌───────────────────────┐
                                                        ▲                              │                       │
                                                        │                              │  syntaxModesRules.ts  │
                                                        │                              │                       │
                                                        │                              └───────────────────────┘
                                          ┌──────────────────────────┐                                          
                                          │                          │                                          
                                          │    testHighlights.py     │                                          
                                          │                          │                                          
                                          └──────────────────────────┘                                          
```